citesphere/src/main/java/edu/asu/diging/citesphere/web/user/AddReferenceController.java

```diff
@@ -0,0 +1,71 @@
+package edu.asu.diging.citesphere.web.user;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.social.zotero.exception.ZoteroConnectionException;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.citesphere.core.exceptions.AccessForbiddenException;
+import edu.asu.diging.citesphere.core.exceptions.CannotFindCitationException;
+import edu.asu.diging.citesphere.core.exceptions.CitationIsOutdatedException;
+import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
+import edu.asu.diging.citesphere.core.exceptions.SelfCitationException;
+import edu.asu.diging.citesphere.core.exceptions.ZoteroHttpStatusException;
+import edu.asu.diging.citesphere.core.exceptions.ZoteroItemCreationFailedException;
+import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.model.bib.ICitation;
+import edu.asu.diging.citesphere.user.IUser;
+
+@Controller
+public class AddReferenceController {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private ICitationManager citationManager;
+
+    @RequestMapping(value = "/auth/group/{zoteroGroupId}/items/{itemId}/addReference", method = RequestMethod.POST)
+    public ResponseEntity<?> addReference(Authentication authentication,
+            @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId,
+            @RequestParam(value = "referenceCitationKey") String referenceCitationKey,
+            @RequestParam(value = "reference") String reference) {
+        try {
+            ICitation citation = citationManager.getCitation((IUser) authentication.getPrincipal(), zoteroGroupId,
+                    itemId);
+            citation = citationManager.addCitationToReferences(citation, referenceCitationKey, reference);
+            citationManager.updateCitation((IUser) authentication.getPrincipal(), zoteroGroupId, citation);
```

Comment:
why does the citation needs to be updated? has anything changed?

Reply:
updateCitation method is called here to update the newly added reference in zetero too. Removing this does not add the reference to zetero's notes.

Reply:
`addCitationToReferences` should save the citation, there shouldn't be an extra call needed (especially not form the controller).

Reply:
still still

Reply:
Moved it to addCitationReference.